docs/proposals/20200804-windows-support.md

```diff
@@ -0,0 +1,324 @@
+
+#### Image Creation
+
+Using cloudbase-init requires the creation of an image with the tooling installed on it since it is not 
+provided out of the box by any cloud providers.  We'll provide packer scripts as part of 
+the [image-builder project](https://github.com/kubernetes-sigs/image-builder) that pre-installs 
+`cloudbase-init`.  It is important to note that while scripts can be provided to build an image, all images 
+built need to adhere to [Windows licensing requirements](https://www.microsoft.com/en-us/licensing/product-licensing/windows-server).
+
+There is prior art for building Windows base images. For example, AKS-Engine has an example implementation for using packer and scripts to do image configuration: https://github.com/Azure/aks-engine/blob/master/vhd/packer/windows-vhd-builder.json.  
+Another example is the the [sig-windows-tools](https://github.com/kubernetes-sigs/sig-windows-tools) which provide scripts for image configuration when using Kubeadm.
+
+Although the Linux implementation in image-builder uses Ansible for configuration, Windows isn't going to share
+the same configuration because [Ansible](https://docs.ansible.com/ansible/latest/user_guide/windows.html) requires [Windows specific modules](https://docs.ansible.com/ansible/latest/modules/list_of_windows_modules.html) to do the configuration. 
+
+#### Kubelet and other component configuration
+
+Due to the lack of privileged containers in Windows, a combination of `PreKubeadmCommands`/`PostKubeadmCommands` 
+scripts and wins.exe can be used to configure the nodes. Wins.exe is currently provided as a way to bootstrap nodes along with kubeadm in the [Kubernetes documentation for adding Windows nodes](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/adding-windows-nodes/).  
+The components from the [preparenode script](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/adding-windows-nodes/#joining-a-windows-worker-node) can be used during image creation.
+
+In the future, when support for [Privileged Containers for Windows containers](https://github.com/kubernetes/enhancements/issues/1981) is merged, we might be able to revisit this proposal 
+and use privileged containers in place of wins.exe enabled containers. 
+
+Each infrastructure providers must provide their own `PreKubeadmCommands`/`PostKubeadmCommands` scripts that
+are required for additional configuration for the node. During planning for Beta we will be able to identify
+common overlapping features that can be added into the the base images in image-builder for re-use.
+
+#### Kubeadm retry support 
+
+Based on the [CAPH Windows implementation](https://github.com/microsoft/cluster-api-provider-azurestackhci), it was 
+identified that the [experimental retry](https://github.com/kubernetes-sigs/cluster-api/pull/2763) for kubeadm 
```

Comment:
I've opened https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1993 to test if it's still needed. Should do the same for the other providers.

Reply:
I've added an issue for Azure: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/998

If the retry support is sufficient, I can remove this section all together.  I still think there could be reason down the line for `osType` on the infra machine, but we can address that later if there is a solid reason.

Reply:
Based on the conversation today at cluster-api meeting, the desire is remove experimental retry for  for v1Alpha4. Since Windows support will be in v1alpha 3 I am going to drop the retry aspect.  We can revisit the `OsType` field if a concrete use case comes up later.